ui/src/app/components/gke-app-configuration-panel.spec.tsx

```diff
@@ -0,0 +1,138 @@
+  AppsApiStub,
+  createListAppsCromwellResponse,
+} from 'testing/stubs/apps-api-stub';
+
+describe(GKEAppConfigurationPanel.name, () => {
+  beforeEach(() => {
+    registerApiClient(AppsApi, new AppsApiStub());
+    notificationStore.set(null);
+
+    serverConfigStore.set({
+      config: {
+        ...defaultServerConfig,
+      },
+    });
+
+    jest
+      .spyOn(appsApi(), 'listAppsInWorkspace')
+      .mockImplementation((): Promise<any> => Promise.resolve([]));
+  });
+
+  const DEFAULT_PROPS = {
+    type: AppType.RSTUDIO,
+    workspaceNamespace: 'aou-rw-1234',
+    ...RSTUDIO_DEFAULT_PROPS,
+  };
+
+  const createWrapper = (
+    propOverrides: Partial<GkeAppConfigurationPanelProps> = {}
+  ) => {
+    const props = {
+      ...DEFAULT_PROPS,
+      ...propOverrides,
+    };
+    return mount(<GKEAppConfigurationPanel {...props} />);
+  };
+
+  it('should show a loading spinner while waiting for the API call to return', async () => {
+    jest
+      .spyOn(appsApi(), 'listAppsInWorkspace')
+      .mockImplementation((): Promise<any> => Promise.resolve([]));
+    const wrapper = createWrapper();
+
+    expect(wrapper.childAt(0).is(Spinner)).toBeTruthy();
+
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(wrapper.childAt(0).is(Spinner)).toBeFalsy();
+  });
+
+  it('should show an error if the API call fails', async () => {
+    jest
+      .spyOn(appsApi(), 'listAppsInWorkspace')
+      .mockImplementation((): Promise<any> => Promise.reject());
+
+    expect(notificationStore.get()).toBeNull();
+
+    const wrapper = createWrapper();
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(notificationStore.get().title).toBeTruthy();
+    expect(notificationStore.get().message).toBeTruthy();
+  });
+
+  it('should not show an error if API call succeeds', async () => {
+    jest
+      .spyOn(appsApi(), 'listAppsInWorkspace')
+      .mockImplementation((): Promise<any> => Promise.resolve([]));
+
+    expect(notificationStore.get()).toBeNull();
+
+    const wrapper = createWrapper();
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(notificationStore.get()).toBeNull();
+  });
+
+  it('should pass the user apps to the specific app component when the API call succeeds', async () => {
+    const workspaceNamespace = 'aou-rw-1234';
+    const apps = [createListAppsCromwellResponse()];
+    const listAppsStub = jest
+      .spyOn(appsApi(), 'listAppsInWorkspace')
+      .mockImplementation((): Promise<any> => Promise.resolve(apps));
+
+    const wrapper = createWrapper({
+      workspaceNamespace,
+      type: AppType.CROMWELL,
+    });
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(listAppsStub).toHaveBeenCalledWith(workspaceNamespace);
+    const cromwellPanel = wrapper.find(CromwellConfigurationPanel);
+    expect(cromwellPanel.exists()).toBeTruthy();
+    expect(cromwellPanel.prop('gkeAppsInWorkspace')).toEqual(apps);
+  });
+
+  it('should display the Cromwell panel when the type is Cromwell', async () => {
```

Comment:
This test and the RStudio test are both new. We weren't testing the `cond` statement before.